config_models/admin.py

```diff
@@ -208,6 +209,5 @@ def edit_link(self, inst):
             return u'--'
         update_url = reverse('admin:{}_{}_add'.format(self.model._meta.app_label, self.model._meta.model_name))
         update_url += "?source={}".format(inst.pk)
-        return u'<a href="{}">{}</a>'.format(update_url, _('Update'))
-    edit_link.allow_tags = True
+        return mark_safe(u'<a href="{}">{}</a>'.format(update_url, _('Update')))
```

Comment:
I think you just want to wrap `mark_safe(u'<a href="{}">{}</a>')` with mark_safe, which will HTML-escape the arguments, which is what we want.

Reply:
Actually, what we want is for `u'<a href="{}">{}</a>' to be marked safe without being escaped, and the arguments to be HTML-escaped.  Another PR with a similar issue used the following solution: https://github.com/edx/edx-platform/blob/f70390d70f3515f5aa59ae8ee6e86212a6e929b1/common/djangoapps/third_party_auth/admin.py#L104
1. I'm not sure if these are equivalent?
2. Not sure if we should prefer one over the other.
A unit test will ensure that the right part is escaped, whatever you use to implement.

Reply:
Yes, @robrap, this is a very good catch. The two solutions you propose are actually equivalent:

```
def format_html(format_string, *args, **kwargs):
    """
    Similar to str.format, but pass all arguments through conditional_escape(),
    and call mark_safe() on the result. This function should be used instead
    of str.format or % interpolation to build up small HTML fragments.
    """
    args_safe = map(conditional_escape, args)
    kwargs_safe = {k: conditional_escape(v) for (k, v) in kwargs.items()}
    return mark_safe(format_string.format(*args_safe, **kwargs_safe))
```

I'll use `format_html`, as it is the cleanest solution of the two. I'll also add a unit test.